app/src/main/java/io/homeassistant/companion/android/sensors/NotificationSensorManager.kt

```diff
@@ -98,6 +99,7 @@ class NotificationSensorManager : NotificationListenerService(), SensorManager {
 
     override fun requestSensorUpdate(context: Context) {
         updateMediaSession(context)
+        updateActiveNotificationCount()
```

Comment:
I noticed that this call was missing here. Maybe it was intentional, but I couldn't come up with an explanation why it wouldn't belong here.

If you think this should be a part of a separate PR, or if it was intentionally not called and I should remove it, please let me know!

Reply:
Did you test this change? Last I checked is only accessible in certain places and may cause an error if not initialized.

Reply:
I remember now `requestSensorUpdate` is not going to make a difference here because we update the sensor when items are added and removed from the status bar. Also we only have access to the status bar when the listener is connected which happens as items get added/removed. I think there might be an old bug on this change.

Reply:
One thing I forgot to mention: without this call, the setting didn't appear until the first notification got posted/removed.

Reply:
yea that also happens for last notification sensors as well, allow list setting wont appear until the code is reachable.

Reply:
`SensorManager::requestSensorUpdate` is called [here](https://github.com/home-assistant/android/blob/15b1518a4466e9c58d26efce7e4605858b88a689/app/src/main/java/io/homeassistant/companion/android/settings/sensor/SensorDetailViewModel.kt#L217) in `SensorDetailViewModel::setEnabled`.

Reply:
> Did you test this change? Last I checked is only accessible in certain places and may cause an error if not initialized.

Thanks for pointing this out. I actually tested it, but only lightly. I tried enabling/disabling the sensor, and triggering sensor updates.

Reply:
I just wanted to point out the `|| !listenerConnected` part in the early return condition 😄

> there may still be times where the listener is not connected and a setting value wont be created

Do you think it is a blocker problem? Eventually the setting will be created, right?

Or I can move the creation of the setting somewhere else where it is guaranteed that it will be called when enabling the sensor. (But I think a proper solution would be to handle the sensor settings more centrally and generally across all sensors.)

Reply:
Its really only these sensors because the code only becomes active when things are posted or removed. The other sensors all provide settings upon being enabled as the code is always reachable. Maybe we should consider a setting initialization step?

we have had that issue with last notification since the setting was introduced so its not a blocker by any means given the amount of time its been like that

Reply:
> Maybe we should consider a setting initialization step?

Yes, that would be a good direction, assuming that we want to keep the current architecture which lists sensor settings from the local DB, and not some statically defined sensor definition. (And yes, I know that the sensor settings list can contain dynamic number of items, like the Last Notification sensor, that's why it would be complicated to refactor)

Reply:
Yea I think if we wanted to do a refactor that would be way out of scope for this PR. Getting the setting to update and appear faster leads to a better user experience IMO. Unless we want to change how it currently is I think it should be ok as we are already checking if the listener is connected for us from the past PR.

Reply:
> Yea I think if we wanted to do a refactor that would be way out of scope for this PR.

I agree!

> Getting the setting to update and appear faster leads to a better user experience IMO.

I agree!

> Unless we want to change how it currently is I think it should be ok as we are already checking if the listener is connected for us from the past PR.

OK, I wouldn't change it now.